### PR TITLE
Introduce black code style

### PR DESCRIPTION
**Pull request by atombrella**
I auto-formatted the code, and included some configuration for it. The same line length as in [`.editorconfig`](https://github.com/certbot/josepy/blob/master/.editorconfig#L14) is used. Most of the changes just change single quotes to double-quotes. All files, including the sphinx configuration, and the example have been reformatted.

I'd prefer that this is merged using [rebase-and-merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#rebase-and-merge-your-pull-request-commits), which would result in the two commits (just like merge pull request, but it doesn't create the merge-commit). The second commit hash can then be included in the [`.git-ignore-revs-file`](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view) which appears to be supported in GitHub now (yeah!).

I think it's a good trial before doing the same for Certbot https://github.com/certbot/certbot/issues/8818 I think @alexzorin would be interested in testing how GitHub works `git blame` with an ignore file.

### CODE REVIEW

File: tests/jwk_test.py [hunk 1]
```diff
@@ -139,101 +151,114 @@ def test_to_partial_json(self):
...
         from josepy.jwk import JWKRSA
-        key = JWKRSA.json_loads("""{
+
+        key = JWKRSA.json_loads(
+            """{
     "kty": "RSA",
     "kid": "bilbo.baggins@hobbiton.example",
     "use": "sig",
-    "n": "n4EPtAOCc9AlkeQHPzHStgAbgs7bTZLwUBZdR8_KuKPEHLd4rHVTeT-O-XV2jRojdNhxJWTDvNd7nqQ0VEiZQHz_AJmSCpMaJMRBSFKrKb2wqVwGU_NsYOYL-QtiWN2lbzcEe6XC0dApr5ydQLrHqkHHig3RBordaZ6Aj-oBHqFEHYpPe7Tpe-OfVfHd1E6cS6M1FZcD1NNLYD5lFHpPI9bTwJlsde3uhGqC0ZCuEHg8lhzwOHrtIQbS0FVbb9k3-tVTU4fg_3L_vniUFAKwuCLqKnS2BYwdq_mzSnbLY7h_qixoR7jig3__kRhuaxwUkRz5iaiQkqgc5gHdrNP5zw",
+    "n": n4EPtAOCc9AlkeQHPzHStgAbgs7bTZLwUBZdR8_KuKPEHLd4rHVTeT-O-XV2jRojdNhxJWTDvNd7nqQ0VEiZQHz_AJmSCpMaJMRBSFKrKb2wqVwGU_NsYOYL-QtiWN2lbzcEe6XC0dApr5ydQLrHqkHHig3RBordaZ6Aj-oBHqFEHYpPe7Tpe-OfVfHd1E6cS6M1FZcD1NNLYD5lFHpPI9bTwJlsde3uhGqC0ZCuEHg8lhzwOHrtIQbS0FVbb9k3-tVTU4fg_3L_vniUFAKwuCLqKnS2BYwdq_mzSnbLY7h_qixoR7jig3__kRhuaxwUkRz5iaiQkqgc5gHdrNP5zw",  # noqa
```
Comment (alexzorin) [inline tests/jwk_test.py:197]: 🤨 I assume (and hope) `black` was not responsible for this quote disappearing, haha.

**Comment by alexzorin** [comment]
I am very excited for GitHub supporting `.git-blame-ignore-revs`!

File: tests/jwk_test.py (hunk 1 above)
Comment (atombrella) [reply tests/jwk_test.py]: Not really sure what happened. I had some issues with flake8, and tried moving the `# noqa` marker. Anyway, it's fixed/reverted now :)

**Comment by atombrella** [comment]
@alexzorin Tests are passing. Except that I might like to change the email address for my user in the two commits (random researchers have started emailing/spamming me to fill in questionnaires for their research, sigh!), at least change-wise, I'm happy. Apologies for force-pushing a few times, but it was needed to ensure the separation of CI configuration, and auto-formatting.

**Comment by bmw** [comment]
Thanks a lot for this @atombrella, but it doesn't look like we have the spare cycles to review this now so I'd like to close this.

Are there significant changes made here that make sense for us to save in a separate branch to use when/if we do decide to tackle this or is it mostly just changes automatically made by `black`?

**Comment by atombrella** [comment]
> Thanks a lot for this @atombrella, but it doesn't look like we have the spare cycles to review this now so I'd like to close this.
> 
> Are there significant changes made here that make sense for us to save in a separate branch to use when/if we do decide to tackle this or is it mostly just changes automatically made by `black`?

There's a bit of configuration for GitHub actions and for `tox.ini` but other than that, it's just a re-format of the whole code base.

**Comment by bmw** [comment]
Got it. Thanks.

Looking at those files, I think it's simple enough to just close this, but if you'd prefer I stash this away in a branch on this repo, please let me know and I can do so.

**Pull request closed without merging**